app.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+# coding=utf-8
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from service import MNIST
+
+PORT_NUMBER = 8080
+
+
+class MyHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        """Handle HTTP GET request.
+
+        If endpoint '/mnist/xxx' where 'xxx' is an integer [0-9999] the result of the prediction will be returned to
+        the user. If not, a helpful message will be displayed detailing the correct endpoint."""
+        mnist_inference = MNIST()
+
+        path_components = self.path.split("/")
+        if path_components[1] == 'mnist':
+
+            try:
```

Comment:
Good to have some error handling here, however, I'm wondering if it's worth considering a simpler or clearer way of managing requests/responses?

Reply:
Almost certainly, I was following the layout of the pytorch-gpu quickstart Terry wrote and propose to leave it as is, because I think it is relatively easy to follow.